src/zrc/api/tests/test_zaken.py

```diff
@@ -469,3 +474,89 @@ def test_sort_startdatum(self):
+        ZDS_CLIENT_CLASS='vng_api_common.mocks.MockClient'
+    )
+    def test_zaak_archiefactiedatum_afleidingswijze_ingangsdatum_besluit(self):
+        zaak = ZaakFactory.create(zaaktype=ZAAKTYPE)
+        zaak_url = reverse('zaak-detail', kwargs={'uuid': zaak.uuid})
+
+        responses = {
+            RESULTAATTYPE: {
+                'url': RESULTAATTYPE,
+                'archiefactietermijn': 'P10Y',
+                'archiefnominatie': Archiefnominatie.blijvend_bewaren,
+                'brondatumArchiefprocedure': {
+                    'afleidingswijze': BrondatumArchiefprocedureAfleidingswijze.ingangsdatum_besluit,
+                    'datumkenmerk': None,
+                    'objecttype': None,
+                    'procestermijn': None,
+                }
+            },
+            STATUSTYPE: {
+                'url': STATUSTYPE,
+                'volgnummer': 1,
+                'isEindstatus': False,
+            },
+            STATUSTYPE2: {
+                'url': STATUSTYPE2,
+                'volgnummer': 2,
+                'isEindstatus': True,
+            },
+            BESLUIT: {
+                'url': BESLUIT,
+                'ingangsdatum': '2020-05-03',
+            }
+        }
+
+        zaakbesluit = ZaakBesluitFactory.create(zaak=zaak, besluit=BESLUIT)
+
+        # Set initial status
+        status_list_url = reverse('status-list')
+        with mock_client(responses):
+            response = self.client.post(status_list_url, {
+                'zaak': zaak_url,
+                'statustype': STATUSTYPE,
+                'datumStatusGezet': isodatetime(2018, 10, 1, 10, 00, 00),
+            })
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, response.content)
+
+        zaak.refresh_from_db()
+        self.assertIsNone(zaak.einddatum)
+
+        # add a result for the case
+        resultaat_create_url = get_operation_url('resultaat_create')
+        data = {
+            'zaak': zaak_url,
+            'resultaattype': RESULTAATTYPE,
+            'toelichting': '',
+        }
+
+        response = self.client.post(resultaat_create_url, data)
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, response.data)
+
+        # Set eindstatus
+        datum_status_gezet = utcdatetime(2018, 10, 22, 10, 00, 00)
+
+        with mock_client(responses):
+            response = self.client.post(status_list_url, {
+                'zaak': zaak_url,
+                'statustype': STATUSTYPE2,
+                'datumStatusGezet': datum_status_gezet.isoformat(),
+            })
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, response.content)
+
+        zaak.refresh_from_db()
+        self.assertEqual(zaak.einddatum, datum_status_gezet.date())
+        self.assertEqual(
+            zaak.archiefactiedatum,
+            datetime.strptime('2020-05-03', '%Y-%m-%d').date() + relativedelta(years=10)
```

Comment:
Be as explicit as possible in tests :-)

```suggestion
            date(2030, 5, 3)  # 2020-05-03 + 10 years
```